presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -168,7 +168,11 @@ public void setQueryTimeout(int seconds)
     public void cancel()
             throws SQLException
     {
-        throw new SQLFeatureNotSupportedException("cancel");
+        checkOpen();
+        ResultSet resultSet = currentResult.get();
```

Comment:
Unfortunately, this won't handle statements like `DROP TABLE`, `CREATE TABLE AS`, etc. I'll add a TODO.